UM/Qt/Bindings/MainWindow.py

```diff
@@ -63,14 +63,22 @@ def __init__(self, parent = None):
                                   int(self._preferences.getValue("general/window_width")),
                                   int(self._preferences.getValue("general/window_height")))
 
-        if not self._preferences.getValue("general/restore_window_geometry"):
+        if (not self._preferences.getValue("general/restore_window_geometry") or
+                self._preferences.getValue("general/last_shutdown_severity") == Application.ShutdownSeverity.HardCrash):
             # Ignore whatever the preferences said.
-            Logger.log("i", "Not restoring window geometry from preferences because 'restore_window_geometry' is false")
+            Logger.log("i", "Window geometry not restored from preferences. "
+                            "See 'general/restore_window_geometry' and/or 'general/last_shutdown_severity' for cause.")
             restored_geometry = QRect(self.DEFAULT_WINDOW_LEFT,
                                       self.DEFAULT_WINDOW_TOP,
                                       self.DEFAULT_WINDOW_WIDTH,
                                       self.DEFAULT_WINDOW_HEIGHT)
 
+        # Now set the 'last shutdown normal' to False, then, when the app closes normally, set it back to True.
+        self._preferences.setValue("general/last_shutdown_severity", Application.ShutdownSeverity.HardCrash)
```

Comment:
This won't work if Cura crashes within 10 seconds, which is the delay for saving the preferences file. But maybe that's acceptable (but I would say no, because many things happen at start and you really don't want to go in a crash/restart loop..)

Reply:
I'll have to rethink this entire feature. I promise I'm not usually this bad... 😅

Reply:
Maybe the preferences should be saved "when appropriate" instead of "regularly". That's what I usually do.